File: HTTPServer.cs
using System;
using System.IO;
using System.Net;
using System.Text;

class SimpleHttpServer
{
    private HttpListener _listener;

    public SimpleHttpServer(string prefix)
    {
        _listener = new HttpListener();
        _listener.Prefixes.Add(prefix);
    }

    public void Start()
    {
        _listener.Start();
        Console.WriteLine("Server started...");
        while (true)
        {
            var context = _listener.GetContext();
            ProcessRequest(context);
        }
    }

    private void ProcessRequest(HttpListenerContext context)
    {
        string filePath = Path.Combine(Directory.GetCurrentDirectory(), context.Request.Url.AbsolutePath.TrimStart('/'));
        if (File.Exists(filePath))
        {
            byte[] fileBytes = File.ReadAllBytes(filePath);
            context.Response.ContentType = "text/html";
            context.Response.ContentLength64 = fileBytes.Length;
            context.Response.OutputStream.Write(fileBytes, 0, fileBytes.Length);
        }
        else
        {
            context.Response.StatusCode = (int)HttpStatusCode.NotFound;
            byte[] errorBytes = Encoding.UTF8.GetBytes("<html><body><h1>404 Not Found</h1></body></html>");
            context.Response.OutputStream.Write(errorBytes, 0, errorBytes.Length);
        }
        context.Response.OutputStream.Close();
    }

    public static void Main(string[] args)
    {
        SimpleHttpServer server = new SimpleHttpServer("http://localhost:8080/");
        server.Start();
    }
}

File: HTTPServer.java
import com.sun.net.httpserver.HttpServer;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;

import java.io.IOException;
import java.io.OutputStream;
import java.nio.file.Files;
import java.nio.file.Paths;

public class SimpleHttpServer {
    public static void main(String[] args) throws IOException {
        HttpServer server = HttpServer.create(new java.net.InetSocketAddress(8080), 0);
        server.createContext("/", new HttpHandler() {
            @Override
            public void handle(HttpExchange exchange) throws IOException {
                String response = new String(Files.readAllBytes(Paths.get("index.html")));
                exchange.sendResponseHeaders(200, response.length());
                OutputStream os = exchange.getResponseBody();
                os.write(response.getBytes());
                os.close();
            }
        });
        server.setExecutor(null);
        server.start();
    }
}
